portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,20 +60,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         // Create ArrayList of comments and populate from the database
         ArrayList<Comment> commentsList = new ArrayList<Comment>();
         Iterator resultIterator = results.asIterable().iterator();
-        int numComments = Integer.parseInt(request.getParameter(NUM_COMMENTS_PARAMETER));
-        if(numComments >= 0 ) {
-            for (int i = 0; i < numComments; i++) {
-                if(resultIterator.hasNext()) {
-                    Entity commentEntity = (Entity) resultIterator.next();
-                    String text = (String) commentEntity.getProperty(TEXT_PROPERTY);
-                    long timestamp = (long) commentEntity.getProperty(TIMESTAMP_PROPERTY);
-                        
-                    Comment comment = new Comment(text, timestamp);
-                    commentsList.add(comment);
-                }
+        String numCommentsString = request.getParameter(NUM_COMMENTS_PARAMETER);
+        if(numCommentsString == "" || numCommentsString == null) {
+            throw new IOException("No value entered for number of comments to display");
```

Comment:
I think BadRequestException [1] might be a better fit for this error.

[1] https://docs.oracle.com/javaee/7/api/javax/ws/rs/BadRequestException.html